core/src/main/java/com/linecorp/armeria/server/PathMappingContextWrapper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static com.linecorp.armeria.server.DefaultPathMappingContext.generateSummary;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.MediaType;
+
+/**
+ * A wrapper class of {@link PathMappingContext}. This would be used to override a parameter of the
+ * existing {@link PathMappingContext} instance.
+ */
+class PathMappingContextWrapper implements PathMappingContext {
+
+    private final PathMappingContext delegate;
+    private List<Object> summary;
+
+    PathMappingContextWrapper(PathMappingContext delegate) {
+        this.delegate = delegate;
```

Comment:
nit: requireNonNull()? I was actually curious about this kinds of package-private class need it? or not?.  I looked up other classes in our repository and found out that some of them have it(like ImmutableHttpParameters) some of them haven't. I think either way is OK, but want to know for consistency of our codes....